### PR TITLE
table: format table error code and error message

### PR DESCRIPTION
**Pull request by imtbkcat**
### What problem does this PR solve? 
Not all the errors in table package have error code.

### What is changed and how it works?
Add error code for these errors and format these error messages.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported variable/fields change

Side effects

Related changes

 - Need to cherry-pick to the release branch

Release note

### CODE REVIEW

File: table/table.go [hunk 1]
```diff
@@ -51,41 +51,41 @@ const (
 
 var (
 	// ErrColumnCantNull is used for inserting null to a not null column.
-	ErrColumnCantNull  = terror.ClassTable.New(codeColumnCantNull, mysql.MySQLErrName[mysql.ErrBadNull])
-	errUnknownColumn   = terror.ClassTable.New(codeUnknownColumn, "unknown column")
-	errDuplicateColumn = terror.ClassTable.New(codeDuplicateColumn, "duplicate column")
+	ErrColumnCantNull  = terror.ClassTable.New(mysql.ErrBadNull, mysql.MySQLErrName[mysql.ErrBadNull])
+	errUnknownColumn   = terror.ClassTable.New(mysql.ErrBadField, mysql.MySQLErrName[mysql.ErrBadField])
+	errDuplicateColumn = terror.ClassTable.New(mysql.ErrFieldSpecifiedTwice, mysql.MySQLErrName[mysql.ErrFieldSpecifiedTwice])
```
Comment (lysu) [inline table/table.go]: maybe need change the caller of `errUnknownColumn`, `errDuplicateColumn` due to "%" changed in message?

File: table/table.go [hunk 2]
```diff
@@ -51,41 +51,41 @@ const (
...
+	ErrColumnStateNonPublic = terror.ClassTable.New(mysql.ErrColumnStateNonPublic, mysql.MySQLErrName[mysql.ErrColumnStateNonPublic])
 	// ErrIndexStateCantNone returns for index none state.
-	ErrIndexStateCantNone = terror.ClassTable.New(codeIndexStateCantNone, "index can not be in none state")
+	ErrIndexStateCantNone = terror.ClassTable.New(mysql.ErrIndexStateCantNone, mysql.MySQLErrName[mysql.ErrIndexStateCantNone])
 	// ErrInvalidRecordKey returns for invalid record key.
-	ErrInvalidRecordKey = terror.ClassTable.New(codeInvalidRecordKey, "invalid record key")
+	ErrInvalidRecordKey = terror.ClassTable.New(mysql.ErrInvalidRecordKey, mysql.MySQLErrName[mysql.ErrInvalidRecordKey])
 	// ErrTruncateWrongValue returns for truncate wrong value for field.
-	ErrTruncateWrongValue = terror.ClassTable.New(codeTruncateWrongValue, "incorrect value")
+	ErrTruncateWrongValue = terror.ClassTable.New(mysql.ErrTruncatedWrongValueForField, mysql.MySQLErrName[mysql.ErrTruncatedWrongValueForField])
```
[the remark this answers is not in the document]
Comment (imtbkcat) [reply table/table.go]: fixed

File: table/table.go (hunk 1 above)
Comment (imtbkcat) [reply table/table.go]: But share same error code may be confused.

**Comment by imtbkcat** [comment]
PTAL @lysu @jackysp

**Pull request merged**